### DISCUSSION TITLE
Subpixel averaging progress loops over and over. Not finishing. 3D simulation.

### DISCUSSION THREAD

**Discussion by ahgibbons** [NanoComp/meep#2064] [Q&A] [answered]
I am trying to make a simulation in 3D, however, the subpixel-averaging step never seems to finish. The percentage increases then loops back to a low percent and repeats. It's not clear to me if this process will finish or not, I left it running for 10 minutes with no apparent progress.

I removed all geometry from the simulation and the same happens.

I reinstalled pymeep with `conda create -n mp -c conda-forge pymeep pymeep-extras` and I have the same result.

Minimum working example is below:

```
import meep as mp

CELL_WIDTH = 200 # nm
CELL_HEIGHT = 200 # nm
CELL_DEPTH  = 200 # nm

LEFT = -CELL_WIDTH/2
CENTER = 0
resolution = 10
dt = 100
runtime = 400 #2000
PML_width = 50.0 # nm
LIGHT_SOURCE_POS = LEFT+PML_width + 50

wavelength = 50

cell = mp.Vector3(CELL_WIDTH, CELL_HEIGHT, CELL_DEPTH)

sources = [mp.Source(mp.ContinuousSource(wavelength=wavelength,width=2),
        component=mp.Ez,
        size=mp.Vector3(1, CELL_HEIGHT, CELL_DEPTH),
        center=mp.Vector3(LIGHT_SOURCE_POS,0,0))]

pml_layers = [mp.PML(PML_width,direction=mp.X),
                mp.PML(PML_width, direction=mp.Y),
                mp.PML(PML_width, direction=mp.Z)]

sim = mp.Simulation(cell_size=cell,
        boundary_layers=pml_layers,
        geometry=[],
        k_point = mp.Vector3(),
        sources=sources,
        resolution=resolution)

 
sim.run(mp.at_beginning(mp.output_epsilon),
    mp.to_appended("ez", mp.at_every(dt, mp.output_efield_z)),
    until=runtime)
```

**Comment by oskooi** [comment, marked as answer]
Since the `geometry` object is empty, the issue here is not subpixel smoothing but rather that you have a 3D cell with (200*10)^3 = 8e9 voxels which will require at least 768 Gb of memory (see this [FAQ](https://meep.readthedocs.io/en/latest/FAQ/#what-is-an-estimate-of-the-memory-requirements-of-a-simulation-given-only-its-cell-size-and-resolution)). Trying to load this much data into memory is why the field initialization stage prior to time stepping is taking so long.

**Reply by ahgibbons** [reply]
Oh..... that was a silly mistake. I got the resolution backwards.
Thank you for the help!
